src/compressed_tensors/quantization/lifecycle/forward.py

```diff
@@ -199,24 +201,40 @@ def _process_quantization(
             sc = scale[:, i].view(-1, 1)
             zp = zero_point[:, i].view(-1, 1) if zero_point is not None else None
 
-            idx = i * group_size
-            if do_quantize:
-                output[:, idx : (idx + group_size)] = _quantize(
-                    x[:, idx : (idx + group_size)],
-                    sc,
-                    zp,
-                    q_min,
-                    q_max,
-                    args,
-                    dtype=dtype,
-                )
-            if do_dequantize:
-                input = (
-                    output[:, idx : (idx + group_size)]
-                    if do_quantize
-                    else x[:, idx : (idx + group_size)]
-                )
-                output[:, idx : (idx + group_size)] = _dequantize(input, sc, zp)
+            if g_idx is not None and i in g_idx:
+                grouped_idx = g_idx == i
```

Comment:
let's rename `i` to make it more explicitly that this is the actual group ID

Reply:
another thing to consider for later on is that since this is really the same between inferences - if we just cache it after the first run to avoid constantly redoing this op